remove platform related code

PPpro opened the pull request:
Re:
https://github.com/cocos-creator/2d-tasks/issues/1360
https://github.com/cocos-creator/2d-tasks/issues/867
https://github.com/cocos-creator/2d-tasks/issues/1305

将引擎 `CC_QQPLAY` `CC_WECHATGAME` 和 `BROWSER_TYPE_WECHATGAME` `BROWSER_TYPE_WECHATGAME_SUB` 相关的代码移进 adapter

关联 pr: 
https://github.com/cocos-creator-packages/weapp-adapter/pull/85
https://github.com/cocos-creator-packages/qqplay-adapter/pull/57
https://github.com/cocos-creator-packages/baidu-adapter/pull/15
https://github.com/cocos-creator-packages/xiaomi-adapter/pull/9

# 05.28 更新（参考第二个 commit）
- 将大块的逻辑里有平台差异的部分抽离出来作为一个独立的适配接口，包括：
```
Audio._createElement()
Audio._touchToPlay()
Audio. _forceUpdatingState()
cc.ContainerStrategy. _setupStyle
```

- 将之前的 **__preAdapter** 更名为 **__globalAdapter**， 在引擎里通过 `__globalAdapter.adaptSys(cc.sys)` 去适配系统信息，除此之外还提供接口：
 ```
adaptView
adaptBrowserGetter
adaptContainerStrategy
```

- 增加 WXSubContextView 和 SwanSubContextView 的模块剔除

- 增加 platform 
```
cc.sys.WECHAT_GAME_SUB
cc.sys.BAIDU_GAME_SUB
```

pandamicro commented on cocos2d/audio/CCAudio.js:
```diff
@@ -149,29 +149,24 @@ Audio.State = {
+            while (item = touchPlayList.pop()) {
+                item.audio.play(item.offset);
+            }
+        });
```
将这一部分代码单独拆分出来作为一个函数：_touchToPlay，然后在小游戏平台中空实现。

这样做的好处是，正常的播放代码维护不会影响到所有小游戏平台的代码

pandamicro commented on cocos2d/audio/CCAudio.js:
```diff
@@ -236,14 +231,13 @@ Audio.State = {
-            this._bindEnded(function () {
-                this._bindEnded();
-            }.bind(this));
-        }
```
同上：_rebindEnded

PPpro replied:
setCurrentTime 接口差异比较大，我整个直接适配了

pandamicro commented on cocos2d/audio/CCAudio.js:
```diff
@@ -418,7 +412,7 @@ let WebAudioElement = function (buffer, audio) {
             let self = this;
             clearTimeout(this._currextTimer);
             this._currextTimer = setTimeout(function () {
-                if (!(CC_QQPLAY || CC_WECHATGAME) && self._context.currentTime === 0) {
```
同上，这个函数主体的代码太多，如果在小游戏平台上都分别实现，会导致维护成本和潜在问题增加

PPpro replied:
WebAudio 不需要适配，微信小游戏上直接用 innerAudioContext
可以看  Audio._createElement 接口

pandamicro commented on cocos2d/core/components/WXSubContextView.js:
```diff
@@ -25,202 +25,191 @@
 
-let WXSubContextView;
-
-if (!CC_EDITOR && !CC_WECHATGAME) {
```
在构件流程中做模块剔除，思考一下不存在组件类型的情况下会不会有额外的问题

pandamicro commented on cocos2d/core/platform/CCInputManager.js:
```diff
@@ -223,15 +223,6 @@ let inputManager = {
      * @return {Object}
      */
     getHTMLElementPosition (element) {
```
将这个函数的返回值控制为返回固定的缓存对象

PPpro replied:
web 平台没办法做这个优化，因为需要考虑 body.scrollTop , 会修改 position 的原始数据，只能每次都创建一份拷贝

小游戏平台可以

pandamicro replied:
那边再额外缓存一个不就好了，这个问题肯定可以解决，想想办法

pandamicro commented on cocos2d/core/platform/CCView.js:
```diff
@@ -1078,16 +1028,7 @@ cc.ContainerStrategy = cc.Class({
-            }
-            // Setup style
-            locContainer.style.width = locCanvas.style.width = w + 'px';
-            locContainer.style.height = locCanvas.style.height = h + 'px';
```
单独写一个 _setupStyle 把，这些代码在 web 端仍然是需要的，不应该删除，在小游戏平台空实现 _setupStyle 就好

PPpro replied:
这部分应该是保留的，我误删了

pandamicro commented on predefine.js:
```diff
@@ -128,6 +128,28 @@ function defineMacro (name, defaultValue) {
+            get: function () {
+                let recommandedUsage;
+                if (name === 'CC_WECHATGAMESUB') {
+                    recommandedUsage = 'cc.sys.browserType === cc.sys.BROWSER_TYPE_WECHAT_GAME_SUB';
```
这个也应该是一个 platform

PPpro replied:
这里我加多一个 platform 字段

PPpro committed:
remove platform related code

remove baidugame platform related code


remove xiaomi platform related code


remove CC_QQPLAY related code


add macro deprecated warning

pandamicro commented on cocos2d/core/components/WXSubContextView.js:
```diff
@@ -132,19 +132,19 @@ else {
             this._stopSubContextMainLoop();
         },
-
+    
```
增加了很多无意义的空行修改，回退掉

PPpro committed:
refine code

PPpro committed:
cache canvas bounding rect

pandamicro commented on cocos2d/core/platform/CCView.js:
```diff
@@ -178,16 +129,31 @@ var View = function () {
+        top: 0,
+        width: 0,
+        height: 0,
+    }
```
完全没必要放在 view 里面啊，就是 inputManager 里面存就好了，view 不关心触点转换的过程，而且这些结构在其他平台也是可以根据 InputManager 实现简化的，但是 view 是通用的。

另外这个结构可以简化为

```
boundingRect = {
    left: 0,
    top: 0,
    adjustedLeft: 0,
    adjustedTop: 0,
    width: 0,
    height: 0
}
```

PPpro committed:
move canvasBoundingRect into inputManager

2youyou2 commented on cocos2d/core/platform/CCView.js:
```diff
@@ -787,6 +736,7 @@ cc.js.mixin(View.prototype, {
         cc.visibleRect && cc.visibleRect.init(this._visibleRect);
 
         renderer.updateCameraViewport();
+        _cc.inputManager._updateCanvasBoundingRect();
```
_cc 是什么？

2youyou2 replied:
如果 cc.inputManager 对外废弃，只限引擎使用的话，应该是改为 cc._inputManager 标明是引擎的私有变量
